graphql/src/main/java/com/scalar/db/graphql/schema/TableGraphQlModel.java

```diff
@@ -0,0 +1,436 @@
+import static graphql.schema.GraphQLTypeReference.typeRef;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+
+import com.google.common.collect.ImmutableSet;
+import com.scalar.db.api.TableMetadata;
+import com.scalar.db.io.DataType;
+import graphql.Scalars;
+import graphql.language.BooleanValue;
+import graphql.schema.GraphQLEnumType;
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLInputObjectType;
+import graphql.schema.GraphQLObjectType;
+import graphql.schema.GraphQLScalarType;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+public class TableGraphQlModel {
+  private static final Set<String> CONSENSUSCOMMIT_ATTRIBUTES =
+      ImmutableSet.of(ID, STATE, VERSION, PREPARED_AT, COMMITTED_AT, CREATED_AT);
+
+  private final String namespaceName;
+  private final String tableName;
+  private final TableMetadata tableMetadata;
+  private final boolean transactionEnabled;
+  private final List<String> fieldNames;
+  private final Map<String, GraphQLScalarType> fieldNameGraphQLScalarTypeMap;
+
+  private final GraphQLObjectType objectType;
+  private final GraphQLInputObjectType primaryKeyInputObjectType;
+  private final GraphQLInputObjectType partitionKeyInputObjectType;
+  private final GraphQLFieldDefinition queryGetField;
+  private final GraphQLInputObjectType getInputObjectType;
+  private final GraphQLObjectType getPayloadObjectType;
+  private final GraphQLFieldDefinition queryScanField;
+  private final GraphQLInputObjectType scanInputObjectType;
+  private final GraphQLEnumType clusteringKeyNameEnum;
+  private final GraphQLInputObjectType clusteringKeyInputObjectType;
+  private final GraphQLInputObjectType orderingInputObjectType;
+  private final GraphQLObjectType scanPayloadObjectType;
+  private final GraphQLFieldDefinition mutationPutField;
+  private final GraphQLInputObjectType putInputObjectType;
+  private final GraphQLInputObjectType putValuesObjectType;
+  private final GraphQLObjectType putPayloadObjectType;
+  private final GraphQLFieldDefinition mutationDeleteField;
+  private final GraphQLInputObjectType deleteInputObjectType;
+  private final GraphQLObjectType deletePayloadObjectType;
+
+  public TableGraphQlModel(String namespaceName, String tableName, TableMetadata tableMetadata) {
+    this.namespaceName = Objects.requireNonNull(namespaceName);
+    this.tableName = Objects.requireNonNull(tableName);
+    this.tableMetadata = Objects.requireNonNull(tableMetadata);
+
+    this.transactionEnabled = tableMetadata.getColumnNames().contains(ID);
+    this.fieldNames =
+        tableMetadata.getColumnNames().stream()
+            .filter(
+                colName ->
+                    !CONSENSUSCOMMIT_ATTRIBUTES.contains(colName)
+                        && !colName.startsWith(BEFORE_PREFIX))
+            .collect(toList());
```

Comment:
Also we can use `ConsensusCommitUtils.removeTransactionalMetaColumns()` here:
```suggestion
    this.fieldNames =
        new ArrayList<>(
            ConsensusCommitUtils.removeTransactionalMetaColumns(tableMetadata).getColumnNames());
```

Reply:
Updated in 389d3c6. Now I'm thinking that it would be better to use a `LinkedHashSet` to hold the `fieldNames` rather than using a list. Do you think it's good to change the type to `LinkedHashSet`?

Reply:
Yes, I think we can change the type to `LinkedHashSet`.